src/modules/ProductLicense.rb

```diff
@@ -1624,6 +1624,59 @@ def beta_seen?(id)
     def beta_seen!(id)
       @beta_file_already_seen[id] = true
     end
+
+    # Search for the product licenses in the addon repository.
+    # @param id [Integer] repository ID
+    # @param tmpdir [String] where to save the licenses
+    # @return [Boolean] true if a license has been found
+    def product_license(id, tmpdir)
+      # make sure the resolvables are loaded into the libzypp pool
+      Pkg.SourceLoad
+      ::FileUtils.mkdir_p(tmpdir)
+
+      products = Pkg.ResolvableProperties("", :product, "")
+      products.reject! { |p| p["source"] != id }
+      product_names = products.map { |p| p["name"] }.uniq
+      log.info("Found products from source #{id}: #{product_names.inspect}")
+      found_license = false
+
+      product_names.each do |product_name|
+        locales = Pkg.PrdLicenseLocales(product_name)
+        log.info("License locales for product #{product_name.inspect}: #{locales.inspect}")
+
+        locales.each do |locale|
+          license = Pkg.PrdGetLicenseToConfirm(product_name, locale)
+          next unless license && !license.empty?
+
+          found_license = true
+          log.info("Found license for language #{locale.inspect}, size: #{license.bytesize} bytes")
+
+          path = File.join(tmpdir, locale.empty? ? "license.txt" : "license.#{locale}.txt")
+          File.write(path, license)
+          log.info("License saved to #{path}")
+        end
+      end
+
+      if found_license
+        @license_dir = tmpdir
+        @license_file_print = repo_license_file(id)
+      end
+
+      found_license
```

Comment:
this means if we found license for at least one product, we are done with licenses?

Reply:
Yes, we can display license only for one product anyway. That's compatible with the old approach, there could be only one `license.tar.gz` file on the medium. No regression.